### PR TITLE
security(rust): Update tokio-tungstenite to fix webpki vuln

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes https://github.com/firezone/firezone/security/dependabot/75
Fixes https://github.com/firezone/firezone/security/dependabot/72

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Lockfile seems to be borked!

**Comment by jamilbk** [comment]
Updated base to #2195 to fix CI

**Comment by jamilbk** [comment]
So it looks like the tokio-tungstenite update somehow broke the ability for the client to connect to the Gateway. It doesn't seem like the portal is returning any available Gateways for the Resource the client is requesting. The Gateway's log [doesn't show any connection attempt](https://github.com/firezone/firezone/actions/runs/6359086598/job/17271925258?pr=2181#step:10:15) from the client.

The client log shows a [ControlProtocolError](https://github.com/firezone/firezone/actions/runs/6359086598/job/17271925258?pr=2181#step:8:43). Unfortunately will require some further debugging.

**Comment by jamilbk** [comment]
So we aren't vulnerable to the `tokio-tungstenite` DoS because we use it for client connections only.

But I discovered why tests are failing: The issue is with our upgrade to `tokio-tungstenite` 0.20 in `phoenix-channel` _only_. Specifically, there seems to be a communication issue between the deployed Relay and the Portal, which causes the client to not find any valid Gateways to connect to.

It looks like our method of sending/receiving messages in `connlib` is unaffected, so I've left this PR open with checks passing to highlight the issue.

Here are the Relay logs for a successful CI integration test (`tokio-tungstenite 0.19`)

https://github.com/firezone/firezone/actions/runs/6372134719/job/17294683583

And here are the Relay logs for an unsuccessful one: (`tokio-tungstenite 0.20`)

https://github.com/firezone/firezone/actions/runs/6371650137/job/17293772652

[The tungstenite `CHANGELOG` for `0.20` mentions some changes around flushing](https://github.com/snapview/tungstenite-rs/blob/master/CHANGELOG.md#0200) -- my first thought is maybe messages aren't getting flushed? And perhaps our use of the `futures` primitives in `connlib` (that we aren't using in the `relay` works around the issue?

cc @thomaseizinger -- this one might be much faster for you to take over from here given the context above.

**Comment by jamilbk** [comment]
Can save the `phoenix-channel` fix for another PR -- this one's ready to go.

File: docker-compose.yml [hunk 1]
```diff
@@ -362,6 +362,7 @@ services:
 
 networks:
   resources:
+    enable_ipv6: false
```
Comment (jamilbk) [inline docker-compose.yml:365]: Disabling IPv6 for resource networks to reduce the number of variables in CI until we add explicit tests for it.

File: rust/phoenix-channel/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ edition = "2021"
 
 [dependencies]
 secrecy = { workspace = true }
-tokio-tungstenite = { version = "0.19.0", features = ["rustls-tls-native-roots"] }
+tokio-tungstenite = { version = "0.19", features = ["rustls-tls-native-roots"] }
```
Comment (jamilbk) [inline rust/phoenix-channel/Cargo.toml:10]: See https://github.com/firezone/firezone/pull/2181#issuecomment-1742149303

**Review by conectado — APPROVED** [review]

**Pull request merged**